src/server/handlers/dynamic.js

```diff
@@ -24,7 +25,26 @@ export default ({ server, config }) => {
       // Set a cache key
       const currentPath = request.url.pathname || '/'
       const isPreview = Boolean(request.query && request.query.tapestry_hash)
-      const cacheKey = normaliseUrlPath(currentPath)
+      let cacheKey = normaliseUrlPath(currentPath)
+
+      // Run cacheKeyHandler function if provided by client
+      const cacheKeyHandler = config.cacheKeyHandler
+      let newCacheKey = cacheKey
+      if (isFunction(cacheKeyHandler)) {
```

Comment:
We've often used `typeof cacheKeyHandler === 'function'` instead of using `lodash`